test/utils.test.js

```diff
@@ -125,3 +126,22 @@ describe('createFetch', () => {
     expect(response.status).toEqual(result.status)
   })
 })
+
+describe('parseRetryAfterHeader', () => {
+  test('numeric retry-after header', () => {
+    const header = '23'
+    expect(parseRetryAfterHeader(header)).toEqual(23000)
+  })
+  test('date retry-after header', () => {
+    const spy = jest.spyOn(global.Date, 'now').mockImplementation(() => new Date('Mon, 13 Feb 2023 23:59:59 GMT'))
+    const header = 'Tue, 14 Feb 2023 00:00:00 GMT'
+    expect(parseRetryAfterHeader(header)).toEqual(1000)
+    expect(spy).toHaveBeenCalled()
+  })
+  test('date retry-after header older than now', () => {
+    const spy = jest.spyOn(global.Date, 'now').mockImplementation(() => new Date('Tue, 14 Feb 2023 00:00:00 GMT'))
+    const header = 'Mon, 13 Feb 2023 23:59:59 GMT'
+    expect(parseRetryAfterHeader(header)).toEqual(0)
+    expect(spy).toHaveBeenCalled()
+  })
+})
```

Comment:
There should be a test for the header value being a non-number and non Date. Does that return 0?

Reply:
If the date is older, it returns zero. (similarly my question above for malformed data, does it return 0).
Returning zero for retryDelay will cause an exception in fetch-retry@3.x: https://github.com/jonbern/fetch-retry/blob/b583df38dd3811e8258e6cd246e93c52c1f44dbb/index.js#LL19C77-L19C77

I would say if:
1. the retry-after date is older than the current date
2. the retry-after date is malformed

--> this situation is as if there is no `Retry-After` header, logically in your code. see `if (retryAfter != null) {` in `HttpExponentialBackoff.js`

Reply:
Thanks for the review! I added the required changes